### ISSUE TITLE
A Wider Scrollbar

### ISSUE THREAD

**Issue by thumos333** [stefan-niedermann/nextcloud-notes#1227]
Hi,

**Is your feature request related to a problem? Please describe.**

The scrollbar in the Notes interface is too narrow for my thumb (or other fingers - I've tried it :) to catch it. It is literally impossible to drag the screen up or down using the scrollbar on the right side of the screen.

**Describe the solution you'd like**

Could the scrollbar be made wider to enable scrolling?

**Additional context**

I am using the latest Notes on Xiaomi's MIUI 10 (based on Android 6).

**Comment by stefan-niedermann** [comment]
Hi @thumos333 

the problem ain't your thumb, the problem is Android. The feature you are looking for is called "**fast scroll**" and allows in list like views to drag the scroll thumb into the opposite direction to quickly scroll the list to this percentual position. 😉 

![Screenshot_20210521-091922_Fennec](https://user-images.githubusercontent.com/4741199/119098199-db39a600-ba15-11eb-94cb-e671f2fe5cf9.png)

This works by definition **not** for text views in Android (because they are not list like) and even for the most list types which are state of the art (like `RecyclerView`s) it needs to be implemented with custom libraries.
I have tried a few in the past as i think that the feature itself is a good idea. Unfortunately most of the libraries i found were poorly documented and not maintained. Including such a dependency is a no-go for me because it will block the further development of the Notes app as `RxMarkdown` did in the past.

I ended up with [`FastscrollEverywhere`](https://github.com/Mixiaoxiao/FastScroll-Everywhere) which also had some serious issues:
- Poor performance on big notes (Some users have notes with 20.000 words)
- It requires a `ScrollView` to be present, which is currently only the case for the preview mode, while the edit mode handles the scrolling itself - and having it only in one mode would be cumbersome
- It clashes with the feature to remember and restore the scroll position of a note

To summarize: Android doesn't support it natively and I haven't been able to find a technologically satisfying way to implement it. Closing it therefore but i will accept a Pull Request if someone has a solution for this.

PS.: If you post a link to the source code of other open source apps which implement this for a text view, it might be possible for contributors to identify and maybe adapt the solutions they chose. I personally have never seen this feature anywhere on a text view 🤷

**Comment by thumos333** [comment]
> If you post a link to the source code of other open source apps which implement this for a text view, it might be possible for contributors to identify and maybe adapt the solutions they chose

Just to make sure - it's about the little bar on the right side of the screen used to scroll the screen up and down with your thumb. Like the one in color green on your screenshot. I can see it in all views in NC Notes but I cannot use it for scrolling.

There is, for example, an open-source notes app called Markor which features a scroll bar: https://github.com/gsantner/markor

**Comment by stefan-niedermann** [comment]
I got it, yes. I can make it 1 cm bug, but this will not make it draggable :wink:.

I'll have a look at the markor implementation, but i doubt that we easily can adapt the solution from there...

**Comment by TtuxX** [comment]
Hi @stefan-niedermann, I hope you are well.
Do you know if there have been useful technological changes since 2021 that could now allow to implement such fast scroll?

I have 1 very long journal note that I now must scroll full speed for around 10 seconds with my thumbs each time I want to add more entries. I am reaching the limit of it and it is a bit difficult now 😅

Many thanks in advance for your kind help if possible :)
